src/Models/Character.php

```diff
@@ -46,4 +63,14 @@ public function getName(): string {
     protected function generateDisplayName() {
         $this->displayName = $this->name;
     }
+    
+    /**
+     * Sets the maximum health of a character to a given value. It also sets the
+     * health if none has been set yet.
+     * @param int $maxhealth
+     */
+    protected function setMaxhealth(int $maxhealth) {
+        $this->maxhealth = $maxhealth;
+        $this->health = $this->health??$this->maxhealth;
```

Comment:
Honestly, I dont think this should do anything w/ health. If I call a mutator on a class w/ an instance variable of the same name, I dont expect it to mutate anything except that instance variable. Seems like a surprising side effect.